### PR TITLE
eww: unstable-2023-08-18 -> 0.5.0

### PR DESCRIPTION
**Pull request by coffeeispower**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Nairou** [comment]
Release notes mentioned building both x11 and wayland support by default now. Is the `withWayland` flag still needed?

**Comment by coffee-is-power** [comment]
Yeah i was thinking the same thing, we should deprecate the eww-wayland package

**Comment by coffee-is-power** [comment]
Actually i will just do that without any waiting for any approval because this is pretty obvious that it is not needed anymore.

File: pkgs/applications/window-managers/eww/default.nix
Comment (adomixaszvers) [inline pkgs/applications/window-managers/eww/default.nix:23]: You might update `cargoHash` if any of cargo dependencies were updated.

File: pkgs/applications/window-managers/eww/default.nix
Comment (adomixaszvers) [inline pkgs/applications/window-managers/eww/default.nix:20]: This should be SRI hash but it looks like "sha256-" + some base32.

**Comment by kirillrdy** [comment]
@coffee-is-power please restore PR template

also consult https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md regarding commit conventions

File: pkgs/applications/window-managers/eww/default.nix
Comment (coffee-is-power) [reply pkgs/applications/window-managers/eww/default.nix:20]: idk how to generate that hash or cargoHash, i just tried using nix-prefetch-url on the tarball

Comment (SuperSandro2000) [reply pkgs/applications/window-managers/eww/default.nix:20]: Set the hash to an empty string and run nix build and the hash will be printed.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25816]: This must be in aliases.nix next to this file

**Comment by coffee-is-power** [comment]
@kirillrdy I'll have to clone the entire repo to change the commit messages :skull: 
imagem

File: pkgs/applications/window-managers/eww/default.nix
Comment (adomixaszvers) [reply pkgs/applications/window-managers/eww/default.nix:20]: Or `nix-shell '<nixpkgs>' -p nix-prefetch-github --command 'nix-prefetch-github elkowar eww --rev v0.5.0'`

File: pkgs/top-level/all-packages.nix
Comment (coffee-is-power) [reply pkgs/top-level/all-packages.nix:25816]: Did not notice that thanks

File: pkgs/applications/window-managers/eww/default.nix
Comment (adomixaszvers) [reply pkgs/applications/window-managers/eww/default.nix:23]: use `lib.fakeHash` as it is written in nixpkgs manual about Rust packaging

File: pkgs/top-level/all-packages.nix
Comment (coffee-is-power) [reply pkgs/top-level/all-packages.nix]: fixed

File: pkgs/applications/window-managers/eww/default.nix
Comment (a-kenji) [reply pkgs/applications/window-managers/eww/default.nix:23]: Yes, it needs to be updated:

> error: build of '/nix/store/qqr5gpd4fjr1pvm3m7zqgqjbck1bz6gl-eww-0.5.0-vendor.tar.gz.drv' on - failed: hash mismatch in fixed-output derivation '/nix/store/qqr5gpd4fjr1pvm3m7zqgqjbck1bz6gl-eww-0.5.0-vendor.tar.gz.drv':
         specified: sha256-4yeu5AgleZMOLKNynGMd0XuyZxyyZ+RmzNtuJiNPN8g=
            got:    sha256-IirFE714NZmppLjwbWk6fxcmRXCUFzB4oxOxBvmYu5U=

**Comment by coffee-is-power** [comment]
So are we now ready to merge?

File: pkgs/applications/window-managers/eww/default.nix
Comment (kirillrdy) [inline pkgs/applications/window-managers/eww/default.nix:19]:
```suggestion
    rev = "v${version}";
```

**Comment by kirillrdy** [comment]
first commit message should be `eww: unstable-2023-08-18 -> 0.5.0` and so should be title of PR

**Review by kirillrdy — APPROVED** [review]
let's give others time to have a look

also small nitpick is to squash related commits ( eg hash updates, or change to wayland attribute alias etc )

File: pkgs/top-level/aliases.nix
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:280]:
```suggestion
  eww-wayland = lib.warn "eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead." eww;
```

**Review by SuperSandro2000 — DISMISSED** [review]

**Comment by a-kenji** [comment]
Result of `nixpkgs-review pr 289595` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eww</li>
  </ul>
</details>

**Review by a-kenji — APPROVED** [review]
LGTM, pending the commit changes already requested.

**Comment by coffee-is-power** [comment]
@a-kenji the commits are good now?

**Comment by a-kenji** [comment]
Yes, LGTM.

Thanks for all the work!

**Comment by coffee-is-power** [comment]
> Yes, LGTM.
> 
> Thanks for all the work!

btw it took ages to do, this repository is so big that my computer was dying to move and squash the commits with lazygit

**Review by adomixaszvers — APPROVED** [review]

**Pull request merged**
